### DISCUSSION TITLE
FEC patch and map update on Audi A4 B9

### DISCUSSION THREAD

**Discussion by Kelmania** [Mr-MIBonk/M.I.B._More-Incredible-Bash#96] [Q&A] [answered]
Hi,

Hope someone could help me here.

I have made a stupid mistake by accidentally formatted my navigation db via MIB , is there anyway to get back the map (or even update the map to 2023 version)?

For starter,
I have upgraded my FW to MHI2_ER_AUG22_K3344_MU1438 (Updated from 2133 --> 2161 --> 3241 --> 3344)
By using MIB 3.4.1, I patched the FEC to 023000EE (I could be wrong here but the log file says so), 
but when I tried to update 2023 maps (P313_N60S5MIBH3_ROW), I run into "No valid update data was found on the media source".

Here is my log file.

[MU1438-JTB-07406.11.1511850CDA.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10469610/MU1438-JTB-07406.11.1511850CDA.log)

Thanks in advance.

**Comment by harman-f** [comment, marked as answer]
In which country do you want to use the navigation?

023000EE FEC is for EU maps only.
You need to add the fitting FEC for your country.

Afterwards you will be able to install new maps.

**Reply by Kelmania** [reply]
I want to use the navigation in Malaysia, i believe the region should be ROW..
i don't see ROW in the FEC list but i believe the region code should be D00 (instead of 000)
should i change the FEC to 023D00EE?

**Reply by harman-f** [reply]
Yes, that should be fine:
https://mibwiki.one/doc/mhi2q-mib-fec-list-j30VUcBAam

Just add it to addFec.txt and run addFec function in GEM again.

**Reply by Kelmania** [reply]
I have just updated to 2023 map with 023D00EE.
It is working flawlessly now.
Thanks for the pointer, much appreciated.

**Reply by Jguog** [reply]
Hi I have the exact same problem you had (Already ran the AIO),. I then added the 023D00EE into the addFec.txt and ran the addFec function in GEM. However, my 023D00EE codes shows up in Red Engineering Menu as illegal. Any ideas?

**Reply by harman-f** [reply]
sure your unit is patched?
send LOG

**Reply by Jguog** [reply]
Thanks. Yes, patched. Log attached below: 
[MU1329-YJL-26019.10.1769690182.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10691099/MU1329-YJL-26019.10.1769690182.log)

**Reply by Jguog** [reply]
I think I figured it out. In my log it says 

`search File MU1329-ifs-root-part2-0x00ba0000-CC05F400.ifs`

`No Flashfiles there!`

But the file name that came in the patch folder was named 

`MU1329-ifs-root-part2-0x00ba0000-C005F400`

If I rename the file it should flash correctly right?

**Reply by Jguog** [reply]
Ok that wasn't the solution. The patch is already done I went to clear the FEC errors and tried adding new FEC again, nothing gets added.
[MU1329-YJL-26019.10.1769690182.log](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/files/10692863/MU1329-YJL-26019.10.1769690182.log)

**Reply by Jguog** [reply]
I have resolved the problem with a fresh copy of M.I.B. 
1. Backed up previous copy of M.I.B from sdcard
2. Format sd card
3. fresh copy of M.I.B and patch files
4. edited the addFEC.txt (023000ee -> 023D00ee)
5. enter GEM -> Backup
6. Run AIO patcher  (it confirmed that patch as been applied already)
7. Run add FEC function
8. Done

**Reply by mustafa-tolu** [reply]
Thanks @Jguog . It worked for us. RoW countries need this. (If this doesn't work check that **Source**: https://mibwiki.one/doc/navigation-gdV1YkhdTO))

**Reply by WalcirW** [reply]
Hi, 
I'm trying to follow the steps but no success on installing updated maps...
What I did: 
1-Installed MIB and are able to enter GEM
2-Downloaded and installed the Patch (my unit is MHI2Q_ER_AUG22_P5092)
3-I've added the suggested key (023d00ee)
4-My unit still has the 023d002a and the new 023d00ee key (not the 023000ee) - I live in Brazil, so should work with the RoW code
5-When I try to update it says my SD doesn't contain valid update files

Any hint or suggestion? Please!

**Reply by mustafa-tolu** [reply]
@WalcirW did you try with a fresh copy? It means using re-formatted SD that is contained re-downloaded files from GitHub repo. You should add the key you need after doing this. And patch your unit. You will see old keys too and new one, it is not weird. 

**Note**: Do not forget to take a backup in the GEM menu if you can.

**Reply by WalcirW** [reply]
Hi @mustafa-tolu, 
I've managed to make it work - not sure if it was a problem with the SD card format (it is a 128Gb unit)...
Thanks for the reply!
